iothub/device/src/InternalClient.cs

```diff
@@ -1502,7 +1499,7 @@ public Task SendEventAsync(string outputName, Message message, CancellationToken
                     throw new ArgumentNullException(nameof(message));
                 }
 
-                message.SystemProperties.Add(MessageSystemPropertyNames.OutputName, outputName);
+                message.SystemProperties[MessageSystemPropertyNames.OutputName] = outputName;
```

Comment:
Using the `Add()` method will fail if the key already exists. Using the array indexer will add or overwrite the value.